rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinUtilsTest.java

```diff
@@ -198,46 +249,121 @@ void getFirstInputSigHash_forP2shP2wshTx_withNoRedeemScript_shouldReturnEmpty()
         btcTx.setWitness(0, witness);
 
         // Act
-        Optional<Sha256Hash> sigHash = getFirstInputSigHash(btcTx);
+        Optional<Script> redeemScript = BitcoinUtils.extractRedeemScriptFromInput(btcTx, FIRST_INPUT_INDEX);
 
         // Assert
-        assertFalse(sigHash.isPresent());
+        assertFalse(redeemScript.isPresent());
     }
 
-    // This test checks that the equals methods of TransactionWitness is the implemented in bitcoinj-thin.
-    // When using the default equals method, the test will fail.
     @Test
-    void equals_withTwoTransactionWitness_withDifferentPushCounts_shouldBeTrue() {
+    void extractRedeemScriptFromInput_withASegWitCompatiblePegoutTx_shouldExtractItProperly() {
         // Arrange
-        TransactionWitness txWitness1 = new TransactionWitness(1);
-        TransactionWitness txWitness2 = new TransactionWitness(2);
+        BtcTransaction btcTx = new BtcTransaction(btcMainnetParams);
+
+        int outputIndex = 0;
+        int nHash = 0;
+        Federation segwitCompatibleFederation = P2shP2wshErpFederationBuilder.builder().build();
```

Comment:
I see these lines are repeated more than once. Should we move them out of the method to be on top at the class level?

Reply:
Another idea would be to move them into a test helper method.

Reply:
done!